core/src/main/java/hudson/model/AllView.java

```diff
@@ -89,13 +116,70 @@ protected void submit(StaplerRequest req) throws IOException, ServletException,
         // noop
     }
 
+    /**
+     * Corrects the name of the {@link AllView} if and only if the {@link AllView} is the primary view and
+     * its name is one of the localized forms of {@link Messages#_Hudson_ViewName()} and the user has opted in to
+     * fixing the view name by setting the system property {@code hudson.mode.AllView.JENKINS-38606} to {@code true}.
+     * Use this method to round-trip the primary view name, e.g.
+     * {@code primaryView = applyJenkins38606Fixup(views, primaryView)}
+     * NOTE: we can only fix the localized name of an {@link AllView} if it is the primary view as otherwise urls
+     * would change, whereas the primary view is special and does not normally get accessed by the
+     * {@code /view/_name_} url.
+     *
+     * @param views the list of views.
+     * @param primaryView the current primary view name.
+     * @return the primary view name - this will be the same as the provided primary view name unless a JENKINS-38606
+     * matching name is detected, in which case this will be the new name of the primary view.
+     * @since FIXME
+     */
+    @Nonnull
+    public static String applyJenkins38606Fixup(@Nonnull List<View> views, @Nonnull String primaryView) {
```

Comment:
Is there a proposed PR calling it? From `cloudbees-folder`? Then you should probably have a clearer name.